zhaquirks/thirdreality/plug.py

```diff
@@ -0,0 +1,123 @@
+"""Third Reality Plug devices."""
+
+from typing import Final
+
+from zigpy.profiles import zha
+from zigpy.quirks import CustomDevice
+import zigpy.types as t
+from zigpy.zcl.clusters.general import (
+    Basic,
+    GreenPowerProxy,
+    Groups,
+    Identify,
+    OnOff,
+    Ota,
+    Scenes,
+)
+from zigpy.zcl.clusters.homeautomation import ElectricalMeasurement
+from zigpy.zcl.clusters.smartenergy import Metering
+from zigpy.zcl.foundation import BaseAttributeDefs, ZCLAttributeDef
+
+from zhaquirks import CustomCluster
+from zhaquirks.const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+from zhaquirks.thirdreality import THIRD_REALITY
+
+THIRD_REALITY_PLUG_CLUSTER_ID = 0xFF03
+RESET_SUMMATION_DELIVERED_ATTR_ID = 0x0000
+
+
+class ControlMode(t.uint8_t):
+    """Reset mode for not clear and clear."""
+
+    NOT = 0
+    CLEAR = 1
+
+
+class ThirdRealityPlugCluster(CustomCluster):
+    """ThirdReality Acceleration Cluster."""
+
+    cluster_id = THIRD_REALITY_PLUG_CLUSTER_ID
+
+    class AttributeDefs(BaseAttributeDefs):
+        """Attribute definitions."""
+
+        reset_summation_delivered: Final = ZCLAttributeDef(
+            id=RESET_SUMMATION_DELIVERED_ATTR_ID,
+            type=ControlMode,
+            is_manufacturer_specific=True,
+        )

```

Comment:
Using a `write_attr_button` from [quirks v2](https://github.com/zigpy/zha-device-handlers/blob/dm/quirks-v2-documentation/quirks_v2.md), a button would be added to HA that would set the attribute to `CLEAR`.
I'm assuming the plug automatically resets it back to `NOT` afterwards?

Reply:
We only need to clear the summation delivered of PLUG to zero when the button is pressed.

Reply:
Hello, is there any case of related projects that you mentioned in this functional code?